init.m

```diff
@@ -23,34 +23,31 @@
 % if installed with robotology superbuild you can directly use
 icubModelsInstallPrefix = getenv('ROBOTOLOGY_SUPERBUILD_INSTALL_PREFIX');
 
-meshFilePrefix = [icubModelsInstallPrefix '/share'];
-% Select the robot using the folder name
-% Example
-robotName = 'iCubGenova04';
-modelPath = [icubModelsInstallPrefix '/share/iCub/robots/' robotName '/'];
-fileName = 'model.urdf';
 %% GENERAL SIMULATION INFO
 % Simulation time and delta_t [s]
 Config.simulationTime = inf;
 Config.GRAVITY_ACC = [0,0,-9.81];
 Config.tStep = 0.001;
 
-% Use OSQP instead of quadprog, typically in the case where the optimization toolbox is not
-% available.
-Config.USE_OSQP = true;
+% Since we cannot assume that the optimization toolbox license is available, we use another QP
+% solver in place of the MATLAB native solver 'quadprog'.
+% The alternate optional solver was the OSQP, but after the changes to support the code generation in
+% the Matlab System block 'step_block', only the QPOASES solver (Simulink block) is supported.
+% So the option below is DEPRECATED.
+Config.USE_OSQP = false;
+Config.USE_QPOASES = true;
 
 % Do you want to enable the Visualizer?
 confVisualizer.visualizeRobot = true;
 
 %% ADD CONFIGURATION FILES
+% Select the robot name from YARP_ROBOT_NAME
+robotName = getenv('YARP_ROBOT_NAME');
+
```

Comment:
Does it make sense to add to the path `src` and `lib` here? I had to add them manually

Reply:
That shouldn't be necessary if you launch the right version of `test_matlab_system_19.mdl` from the repo root path. When the model is loaded, the `PostLoadFcn` function runs the following commands:
```
% Set path to the utility functions and to WBC library
import wbc.*
addpath(genpath('./src'));
addpath('./lib/RobotDynamicsWithContacts');
```
which adds all the required paths, including those you mentioned. for some reason you're loading the wrong mdl version. Try starting MATLAB from scratch...